packages-web/maps/src/components/LeafletMap.tsx

```diff
@@ -1,167 +1,101 @@
-import { createElement, ReactElement, useEffect, useRef } from "react";
-import {
-    FeatureGroup,
-    LatLngLiteral,
-    Map,
-    MapOptions,
-    Marker as LeafletMarker,
-    icon,
-    tileLayer,
-    TileLayerOptions,
-    TileLayer
-} from "leaflet";
+import { createElement, ReactElement, useEffect, useRef, useState } from "react";
+import { Map, Marker as MarkerComponent, Popup, TileLayer } from "react-leaflet";
 import classNames from "classnames";
 import { Marker, SharedProps } from "../../typings/shared";
 import { MapProviderEnum } from "../../typings/MapsProps";
 import { getDimensions } from "../utils/dimension";
-import { Option } from "../utils/data";
-import { customUrls, mapAttr } from "../utils/leaflet";
+import { translateZoom } from "../utils/zoom";
+import { latLngBounds } from "leaflet";
+import { baseMapLayer } from "../utils/leaflet";
 
 export interface LeafletProps extends SharedProps {
     mapProvider: MapProviderEnum;
     attributionControl: boolean;
 }
 
 export function LeafletMap(props: LeafletProps): ReactElement {
+    const [isLoaded, setIsLoaded] = useState(false);
+    const center = { lat: 51.906688, lng: 4.48837 };
     const map = useRef<Map>();
-    const leafletDivNode = useRef<HTMLDivElement>(null);
-    const defaultCenterLocation: LatLngLiteral = { lat: 51.906688, lng: 4.48837 };
-    const markerGroup = useRef<FeatureGroup>(new FeatureGroup());
+    const {
+        autoZoom,
+        attributionControl,
+        optionScroll: scrollWheelZoom,
+        optionZoomControl: zoomControl,
+        zoomLevel: zoom,
+        optionDrag: dragging
+    } = props;
 
     useEffect(() => {
-        const mapOptions: MapOptions = {
-            scrollWheelZoom: props.optionScroll,
-            zoomControl: props.optionZoomControl,
-            attributionControl: props.attributionControl,
-            zoom: props.zoomLevel,
-            minZoom: 1,
-            maxZoom: 18, // Max zoom for HERE, OSM & MapBox
-            dragging: props.optionDrag,
-            center: defaultCenterLocation,
-            closePopupOnClick: false
-        };
-        if (leafletDivNode.current && !map.current) {
-            map.current = new Map(leafletDivNode.current, mapOptions)
-                .addLayer(baseMapLayer(props.mapProvider, props.mapsToken))
-                .addLayer(markerGroup.current);
-        }
-
-        return () => {
-            if (map.current) {
-                map.current.off();
-                map.current.remove();
-            }
-        };
-    }, []);
-
-    useEffect(() => {
-        if (map.current) {
-            addMarkers(
-                map.current,
-                markerGroup.current,
-                props.locations,
-                props.currentLocation,
-                props.zoomLevel,
-                props.autoZoom
+        if (map.current && isLoaded) {
```

Comment:
Do you need `isLoaded` check? Because theoretically `map.current` won't be there before you are done loading anyway.

Reply:
Done